interbtc-spec/docs/source/spec/oracle.rst

```diff
@@ -3,137 +3,122 @@
+``hour``                Inclusion estimate for 6 block (~60 min) inclusion.
+======================  ========================================================================
 
-.. _btcTxFeesPerByte:
 
-BtcTxFeesPerByte
-................
+Scalars
+-------
 
-The estimated inclusion time for a Bitcoin transaction MUST be stored in Satoshis per byte.
+.. _MaxDelay:
 
-.. tabularcolumns:: |l|l|L|
+MaxDelay
+........
 
-=========================  ==================  ========================================================
-Parameter                  Type                Description
-=========================  ==================  ========================================================
-``fast``                   u32                 The fee to include a BTC transaction within the next block.
-``half``                   u32                 The fee to include a BTC transaction within the next three blocks (~30 min).
-``hour``                   u32                 The fee to include a BTC transaction within the six blocks  (~60 min).
-=========================  ==================  ========================================================
+The time after which a reported value will no longer be considered valid.
 
 
 Maps
 ----
 
+Aggregate
+.........
+
+Maps ``oracleKey`` to the median of all unexpired values reported by oracles for that key.
+
 AuthorizedOracles
 .................
 
 The account(s) of the oracle. Returns true if registered as an oracle.
 
+ValidUntil
+..........
 
-Functions
-~~~~~~~~~
-
-.. _setExchangeRate:
-
-setExchangeRate
----------------
-
-This function sets the latest base exchange rate.
-
-Specification
-.............
-
-*Function Signature*
-
-``setExchangeRate(oracleId, rate)``
+Maps OracleKeys to a timestamp that indicates when one of the values expires, at which time a new aggregate needs to be calculated.
 
-*Parameters*
+RawValues
+.........
 
-* ``oracleId``: the oracle account calling this function.
-* ``rate``: the fixed point exchange rate.
+Maps OracleKeys and account ids to raw timestamped values. 
 
-*Events*
+RawValuesUpdated
+................
 
-* :ref:`setExchangeRateEvent`
+Maps OracleKey to a boolean value that indicates that a new value has been received that has not yet been included in the aggregate.
 
-*Preconditions*
+AuthorizedOracles
+.................
 
-* The function call MUST be signed by ``oracleId``.
-* The BTC Parachain status in the :ref:`security` component MUST NOT be ``SHUTDOWN:2``.
-* The oracle MUST be authorized.
+Maps oracle ``accountId`` to the oracle's name. The presence of an account id in this map indicates that the account is authorized to feed values.
 
-*Postconditions*
 
-* The ``ExchangeRate`` MUST be set to the provided ``rate``.
-* The ``LastExchangeRateTime`` MUST be updated to the current time.
-* If the status in :ref:`security` is ``ERROR:1``, the system MUST be set to ``RUNNING:0``.
+Functions
+~~~~~~~~~
 
-.. _setBtcTxFeesPerByte:
+.. _feedValues:
 
-setBtcTxFeesPerByte
```

Comment:
Not sure what benefit we can from changing the events actually since it doesn't affect the function calling - but does require a re-write in stats.

Reply:
It was an attempt to keep the API as close as possible to orml-oracle. While it is unlikely that oracle providers depend on the events, it is not impossible, so I chose to change it